backend/src/main/java/com/github/wekaito/backend/ProfileService.java

```diff
@@ -40,11 +42,33 @@ Card[] fetchCards(Optional<String> name, Optional<String> color, Optional<String
         return Objects.requireNonNull(response).getBody();
     }
 
-    public void addDeck(Deck deck) {
-        this.deckRepo.save(deck);
+    public void addDeck(DeckWithoutId deckWithoutId) {
+        Deck deckToSave = new Deck(
+                idService.createId(),
+                deckWithoutId.name(),
+                deckWithoutId.cards()
+        );
+        this.deckRepo.save(deckToSave);
     }
 
     public Deck[] getDecks() {
         return this.deckRepo.findAll().toArray(new Deck[0]);
     }
+
+    public void updateDeck(String id, DeckWithoutId deckWithoutId) {
+        Deck deckToSave = new Deck(
+                id,
+                deckWithoutId.name(),
+                deckWithoutId.cards()
+        );
+        this.deckRepo.save(deckToSave);
+    }
+
+    public Deck getDeckById(String id) {
+        return this.deckRepo.findById(id).orElseThrow();
+    }
+
+    public void deleteDeck(String id) {
+        this.deckRepo.deleteById(id);
```

Comment:
on this point you could check if the deck with this id exist:
```java
if (!deckRepo.existsById(id)) throw new IllegalArgumentException();
```
It could also be a good test case